File: src/notify-sms/handler.rb
require 'twilio-ruby'

def handle(event:, context:)
  @client = Twilio::REST::Client.new ENV["TWILIO_ACCOUNT_SID"], ENV["TWILIO_AUTH_TOKEN"]

  message = "Hey #{event["participant"]["name"]}, you're Secret Santa for #{event["recipient"]["name"]} this year!"

  @client.messages.create(
    from: ENV["SMS_FROM"],
    to: event["participant"]["number"],
    body: message
  )
end

File: src/notify-email/handler.py
import os
import requests


def handle(event, context):
    message = "Hey {}, you're Secret Santa for {} this year!".format(
        event["participant"]["name"], event["recipient"]["name"]
    )

    requests.post(
        "https://api.eu.mailgun.net/v3/" + os.environ["MAILGUN_DOMAIN"] + "/messages",
        auth=("api", os.environ["MAILGUN_API_KEY"]),
        data={
            "from": os.environ["EMAIL_FROM"],
            "to": event["participant"]["email"],
            "subject": "Who are you Secret Santa for?",
            "text": message,
        },
    )
